homeassistant/components/mobile_app/__init__.py

```diff
@@ -1,355 +1,80 @@
-        ATTR_MODEL: device[ATTR_MODEL],
-        ATTR_OS_VERSION: device[ATTR_OS_VERSION],
-        ATTR_SUPPORTS_ENCRYPTION: device[ATTR_SUPPORTS_ENCRYPTION],
-    }
-
-
-def register_device_webhook(hass: HomeAssistantType, store, device):
-    """Register the webhook for a device."""
-    device_name = 'Mobile App: {}'.format(device[ATTR_DEVICE_NAME])
-    webhook_id = device[CONF_WEBHOOK_ID]
-    webhook.async_register(hass, DOMAIN, device_name, webhook_id,
-                           partial(handle_webhook, store))
+REQUIREMENTS = ['PyNaCl==1.3.0']
 
 
-async def async_setup(hass, config):
+async def async_setup(hass: HomeAssistantType, config: ConfigType):
     """Set up the mobile app component."""
-    conf = config.get(DOMAIN)
-
     store = hass.helpers.storage.Store(STORAGE_VERSION, STORAGE_KEY)
     app_config = await store.async_load()
     if app_config is None:
-        app_config = {}
-
-    hass.data[DOMAIN] = app_config
+        app_config = {ATTR_DELETED_IDS: [], ATTR_REGISTRATIONS: {}}
 
-    for device in app_config.values():
-        register_device_webhook(hass, store, device)
+    if hass.data.get(DOMAIN) is None:
+        hass.data[DOMAIN] = {ATTR_DELETED_IDS: [], ATTR_REGISTRATIONS: {}}
 
-    if conf is not None:
-        hass.async_create_task(hass.config_entries.flow.async_init(
-            DOMAIN, context={'source': config_entries.SOURCE_IMPORT}))
+    hass.data[DOMAIN][ATTR_DELETED_IDS] = app_config[ATTR_DELETED_IDS]
+    hass.data[DOMAIN][ATTR_REGISTRATIONS] = app_config[ATTR_REGISTRATIONS]
+    hass.data[DOMAIN][ATTR_STORE] = store
 
-    hass.http.register_view(DevicesView(store))
-
-    return True
+    for device in app_config[ATTR_REGISTRATIONS].values():
+        setup_device(hass, store, device)
 
+    register_http_handlers(hass, store)
+    register_websocket_handlers(hass)
 
-async def async_setup_entry(hass, entry):
-    """Set up an mobile_app entry."""
     return True
 
 
-class DevicesView(HomeAssistantView):
-    """A view that accepts device registration requests."""
+@bind_hass
+def async_devices(hass: HomeAssistantType, component: str) -> list:
+    """Return all devices with the given domain set for app_component."""
+    devices = []
 
-    url = '/api/mobile_app/devices'
-    name = 'api:mobile_app:register-device'
+    if DOMAIN not in hass.data:
+        return devices
 
-    def __init__(self, store):
-        """Initialize the view."""
-        self._store = store
+    for device in hass.data[DOMAIN][ATTR_REGISTRATIONS].values():
+        if device.get(ATTR_APP_COMPONENT) == component:
+            devices.append(device)
 
-    @RequestDataValidator(REGISTER_DEVICE_SCHEMA)
-    async def post(self, request, data):
-        """Handle the POST request for device registration."""
-        hass = request.app['hass']
+    return devices
 
-        resp = {}
 
-        webhook_id = generate_secret()
+@bind_hass
+def async_devices_with_key(hass: HomeAssistantType, component: str,
```

Comment:
I think that this method is overkill. This can done inside calling components.

Reply:
Removed for a later PR